core/server/data/migrations/versions/4.17/02-add-offers-table.js

```diff
@@ -0,0 +1,17 @@
+const utils = require('../../utils');
+
+module.exports = utils.addTable('offers', {
+    id: {type: 'string', maxlength: 24, nullable: false, primary: true},
+    name: {type: 'string', maxlength: 191, nullable: false},
+    slug: {type: 'string', maxlength: 191, nullable: false, unique: true},
+    product_id: {type: 'string', maxlength: 24, nullable: false},
+    interval: {type: 'string', maxlength: 50, nullable: false, validations: {isIn: [['monthly', 'yearly']]}},
+    currency: {type: 'string', maxlength: 50, nullable: false},
+    discount_type: {type: 'string', maxlength: 50, nullable: false},
+    discount_amount: {type: 'integer', nullable: false},
+    duration: {type: 'string', maxlength: 50, nullable: false},
```

Comment:
If we have a duration of `'repeating'` then we need to store for how long it repeats (e.g. 3 months) so I think we either need another column of like `duration_count` or something or we store a number in this column 🤔 

`0` - once
`Infinity` - forever
`1`,`2`,...`1000` - number of months

Reply:
🤔 Good catch, i missed the need for storing `duration_count` in case of `repeating`, i'll add a nullable `duration_in_months ` integer column that can store the duration count in number of months to stay consistent with stripe, don't think overloading the `duration` column with number is good

Reply:
Added `duration_in_months` column now